src/control/security/auth/auth_sys.go

```diff
@@ -46,43 +46,63 @@ type UserExt interface {
 	LookupGroupID(gid uint32) (*user.Group, error)
 }
 
-// HashFromToken will return a SHA512 hash of the token data
-func HashFromToken(token *auth.Token) ([]byte, error) {
-	// Generate our hash (not signed yet just a hash)
-	hash := sha512.New()
-
+//VerifierFromToken will return a SHA512 hash of the token data. If a signing key
```

Comment:
The comment seems to indicate that a nil signing key is possible. Is that accurate?

Reply:
This is correct. If you pass a nil signing key you just get the hash back.

Reply:
Hmm. What are the implications for a caller accidentally passing in a nil key? Could they get back a blob that they think is signed but actually isn't?

It seems like it would be safer to separate these into two functions, with one just returning the hash of the token data and the other one asserting that the returned blob is signed because the key wasn't nil.

Reply:
If the caller passed in a nil key as per the comment they should expect it to just return the hash. That's why its VerifierFromToken and not SignatureFromToken.

Reply:
But imagine that the caller is supplying the key from a struct member as part of some larger operation. Also imagine that someone other than you is writing this calling code and so they're not as attuned to the subtleties here. So they inadvertently pass in a nil key and they get back what they think is a signature but it's just the token hash. What happens then?

We should try to avoid these kind of ambiguous situations in API surface. For internal functions it's sometimes ok to do double-duty based on whether or not a parameter is nil, but for public stuff it'd be much better (if somewhat more verbose) to have functions which do one thing and don't accept optionally-nil parameters.

Reply:
Generally don't like to fall back on an argument from authority, but I was just reading through Dave Cheney's guide and saw this: https://dave.cheney.net/practical-go/presentations/gophercon-singapore-2019.html#_rather_than_commenting_a_block_of_code_refactor_it